### PR TITLE
hotfixes for device registry

### PR DESCRIPTION
**Pull request by Baalmart**
**_WHAT DOES THIS PR DO?_**

- [x] `network` field is mandatory during device creation
- [ ] fix runtime errors for create-photos usecase. Something strange in STAGING env!
- [x] Solves issue which happens on querying events for a non existent site, `pages` are equate to `null`
- [x] do some QA for Kafka broker in device registry....focusing mainly on STAGING.
- [x] Only run Kafka consumer in STAGING, temporarily disable PRODUCTION
- [x] Make the error log for app ENTRY more descriptive in Slack notifs channel. Cases where someone accesses a non-existent endpoint
- [x] Using ENVs for the valid tenants/networks/organisations during input validations.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [AN-267]

**_HOW DO I TEST OUT THIS PR?_**
```
cd src/device-registry
npm install
npm run stage-mac

```
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [create device](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-3be73eff-b260-4c5c-933d-5f193ee89cf5)
- [x] [SOFT add device](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-2341976e-cf2c-4b32-ba81-110693e47d44)
- [ ] [create photos](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-333b6062-591b-430d-a8e1-3c434c14d354)
- [ ] [update photos](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-da49f3c6-858f-4571-8404-421e5ffbc56e)
- [ ] [delete photos](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-a0e42427-2789-4468-a96d-a698fe8423e8)
- [ ] [view photos](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-3023d862-3b0a-4b0d-a3fb-b59917dc73e6)

[AN-267]: https://airqoteam.atlassian.net/browse/AN-267?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/device-registry/.gitignore [hunk 1]
```diff
@@ -6,4 +6,5 @@ log
 scripts/data-airqlouds.js
 scripts/data-sites.js
 scripts/data-devices.js
-scripts/data-activities.js
\ No newline at end of file
+scripts/data-activities.js
```
Comment (Baalmart) [inline src/device-registry/.gitignore:9]: just excluding some new testing/script files

File: src/device-registry/app.js [hunk 2]
```diff
@@ -35,6 +36,13 @@ const runKafkaConsumer = async () => {
     await kafkaConsumer.run({
       eachMessage: async ({ message }) => {
         const measurements = JSON.parse(message.value).data;
+        if (!Array.isArray(measurements) || isEmpty(measurements)) {
+          logger.error(
```
Comment (Baalmart) [inline src/device-registry/app.js:40]: specifically tagging KAFKA log messages accordingly

File: src/device-registry/app.js [hunk 3]
```diff
@@ -49,14 +57,17 @@ const runKafkaConsumer = async () => {
       },
     });
   } catch (error) {
-    logElement("KAFKA CONSUMER RUN ERROR", error.message);
+    logger.error("KAFKA: internal server error", error.message);
```
Comment (Baalmart) [inline src/device-registry/app.js:60]: specifically tagging KAFKA log messages accordingly

File: src/device-registry/app.js [hunk 4]
```diff
@@ -49,14 +57,17 @@ const runKafkaConsumer = async () => {
...
   }
 };
 
-runKafkaConsumer();
+if (constants.ENVIRONMENT === "STAGING ENVIRONMENT") {
```
Comment (Baalmart) [inline src/device-registry/app.js:64]: R KAFKA consumer in STAGING environment ONLY, for now...... @NoahNsimbe / @123MwanjeMike

Comment (Baalmart) [inline src/device-registry/app.js:64]: R KAFKA consumer in STAGING environment ONLY, for now...... @NoahNsimbe / @123MwanjeMike

File: src/device-registry/app.js [hunk 5]
```diff
@@ -93,76 +104,70 @@ app.use(function(req, res, next) {
...
     res.status(err.status).json({
       success: false,
       message: "bad request error",
       errors: { message: err.message },
     });
-  }
-
-  if (err.status === 401) {
+  } else if (err.status === 401) {
+    logger.error(`unauthorized--- ${err.message}`);
```
Comment (Baalmart) [inline src/device-registry/app.js:121]: adding more descriptive messages for our logger

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @Baalmart 
LGTM

**Pull request merged**